test/Classes/Bar.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LaminasTest\Di\Classes;
+
+use Countable;
+use Iterator;
+
+class Bar
```

Comment:
Let's call this `IntersectionTypeConstructorDependency`

Reply:
Im wondering if all tools can correctly deal with the & in the intersection type hint though.

Reply:
A general dependency upgrade could be introduced here: most static analysis tooling is already able to deal with it. PHPCS is probably the one that is more problematic.